### ISSUE TITLE
virtualenv.in-project also changes behavior c.f. multiple python versions

### ISSUE THREAD

**Issue by marius311** [python-poetry/poetry#7451]
- **Poetry version**: 1.3.2
- **Python version**: 3.9.16 and 3.10.6
- **OS version and name**: Ubuntu 22.04

- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue

The following was an unexpected and undesired (for me) side-effect of changing `virtualenvs.in-project`. 

If `virtualenvs.in-project=true`, then the virtual environment is locked into Python version:

```
$ python3.9 -m poetry config virtualenvs.in-project true
$ python3.9 -m poetry init -n --python ^3.9
$ python3.9 -m poetry shell
Creating virtualenv test in ~/test/.venv
Spawning shell within ~/test/.venv
$ . ~/test/.venv/bin/activate
(test-py3.9) $ 
exit
$ python3.10 -m poetry shell
Spawning shell within ~/test/.venv
$ . ~/test/.venv/bin/activate
(test-py3.9) $ 
```

But if I change to `virtualenvs.in-project=false` and start from scratch:
```
$ python3.9 -m poetry config virtualenvs.in-project false
$ python3.9 -m poetry init -n --python ^3.9
$ python3.9 -m poetry shell
Creating virtualenv test-c_b_njw9-py3.9 in ~/.cache/pypoetry/virtualenvs
Spawning shell within ~/.cache/pypoetry/virtualenvs/test-c_b_njw9-py3.9
$ . ~/.cache/pypoetry/virtualenvs/test-c_b_njw9-py3.9/bin/activate
(test-py3.9) $ 
exit
$ python3.10 -m poetry shell
Creating virtualenv test-c_b_njw9-py3.10 in ~/.cache/pypoetry/virtualenvs
Spawning shell within ~/.cache/pypoetry/virtualenvs/test-c_b_njw9-py3.10
$ . ~/.cache/pypoetry/virtualenvs/test-c_b_njw9-py3.10/bin/activate
(test-py3.10) $ 
exit
```

The active environment tracks the Python that launched poetry. 

1. I'm not trying to say anything about which behavior is right, but the main issue I want to raise here is just that its really unintuitive and prone to errors to have this behavior depend on the (to me) seeminly unrelated `virtualenvs.in-project` option, which instead I thought dealt with simply where files are stored. 

2. That said, as a secondary thing, I do find the first behavior much preferred. To me it seems much more useful to have a virtual environment also locked to a Python version, and unless I say otherwise, I get _that_ version of Python in the shell, regardless what outer Python might have spawned it. I mean, all the packages are pinned to specific versions, doesn't it make sense to pin Python itself too? 

But regardless if there's disagreement on 2, I think 1 is much less controversial and could hopefully be fixed to have them be the same behavior, whatever that might be.

**Comment by finswimmer** [comment]
Hey @marius311,

this is an expected behavior. As long as `virtualenvs.in-project` is not explicit set to `false` Poetry will activate the virtualenvironment found at `.venv` if there is such a folder. (https://python-poetry.org/docs/configuration/#virtualenvsin-project).

If `virtualenvs.in-project` is not set to `true` and there is no `.venv` Poetry will create and activate a venv with the interpreter it was installed with. 

You invoke Poetry as a module (which is not the intended way by the way). One time with python3.9 and one time with python3.10. So I guess you've installed Poetry twice?

**Comment by marius311** [comment]
Thanks for the quick reply. 

> You invoke Poetry as a module (which is not the intended way by the way). One time with python3.9 and one time with python3.10. So I guess you've installed Poetry twice?

Yea, I wrote that that way just for the demonstration. The realistic thing (how I ended up here) is that I'm developing for multiple different versions and I might be switching around Python versions, or I'm on a cluster and I `module swap python/3.9 python/3.10` etc..., basically I might not always know exactly which Python I get when I type `$ poetry`. I _thought_ that was fine to not know that (in fact, that was a big draw of poetry) because I thought each folder with a `pyproject/toml/poetry.lock` was a pristine reproducible unit, but I have now realized whether that's true or not depends on `virtualenvs.in-project`. 

So to phrase the problem another way, `virtualenvs.in-project` changes two things 1) this aforementioned behavior and 2) where files are stored. The combination of options I actually want (always same Python version + centralized file storage) is not available. It would make more sense to me to have two indepdendent options.  I do see `virtualenvs.prefer-active-python` but afaict that only affects environment creation. 

Fwiw, I'd read those docs you linked many times and had never realized this. I would say at the very least they could be improved to make this distinction clearer.
